src/AutoRest.CSharp/Common/AutoRest/Plugins/Internalizer.cs

```diff
@@ -0,0 +1,378 @@
+            }
+
+            public override SyntaxNode? VisitStructDeclaration(StructDeclarationSyntax node)
+            {
+                node = (StructDeclarationSyntax)base.VisitStructDeclaration(node)!;
+
+                if (IsPublic(node.Modifiers) && !_modelsToKeep.Contains(node.Identifier.ToString()))
+                {
+                    ModelDeclarations = ModelDeclarations.Add(node);
+                }
+                return node;
+            }
+
+            public override SyntaxNode? VisitEnumDeclaration(EnumDeclarationSyntax node)
+            {
+                node = (EnumDeclarationSyntax)base.VisitEnumDeclaration(node)!;
+                if (IsPublic(node.Modifiers) && !_modelsToKeep.Contains(node.Identifier.ToString()))
+                {
+                    ModelDeclarations = ModelDeclarations.Add(node);
+                }
+                return node;
+            }
+        }
+
+        private class MemberVisitor : CSharpSyntaxRewriter
+        {
+            private List<SyntaxNode> _members = new();
+
+            public IEnumerable<SyntaxNode> PublicMembers => _members;
+
+            private Func<SyntaxTokenList, bool> _predicate;
+
+            public MemberVisitor(Func<SyntaxTokenList, bool> accessibility)
+            {
+                _predicate = accessibility;
+            }
+
+            public override SyntaxNode? VisitClassDeclaration(ClassDeclarationSyntax node)
+            {
+                node = (ClassDeclarationSyntax)base.VisitClassDeclaration(node)!;
+                if (_predicate(node.Modifiers))
+                {
+                    _members.Add(node);
+                }
+                return node;
+            }
+
+            public override SyntaxNode? VisitPropertyDeclaration(PropertyDeclarationSyntax node)
+            {
+                node = (PropertyDeclarationSyntax)base.VisitPropertyDeclaration(node)!;
+                if (_predicate(node.Modifiers))
+                {
+                    _members.Add(node);
+                }
+                return node;
+            }
+
+            public override SyntaxNode? VisitMethodDeclaration(MethodDeclarationSyntax node)
+            {
+                node = (MethodDeclarationSyntax)base.VisitMethodDeclaration(node)!;
+                if (_predicate(node.Modifiers))
+                    _members.Add(node);
+                return node;
+            }
+
+            public override SyntaxNode? VisitConstructorDeclaration(ConstructorDeclarationSyntax node)
+            {
+                node = (ConstructorDeclarationSyntax)base.VisitConstructorDeclaration(node)!;
+                if (_predicate(node.Modifiers))
+                    _members.Add(node);
+                return node;
+            }
+        }
+
+        private static bool IsPublic(SyntaxTokenList tokenList)
+            => tokenList.Any(token => token.IsKind(SyntaxKind.PublicKeyword));
+
+        private static bool IsStatic(SyntaxTokenList tokenList)
```

Comment:
Is there any reason that we need to keep this method? I don't see any reference here.

Reply:
Let me remove this